diff --git a/openscholar/modules/os/modules/os_restful/plugins/restful/node/biblio/1.0/BiblioNodeRestfulBase.class.php b/openscholar/modules/os/modules/os_restful/plugins/restful/node/biblio/1.0/BiblioNodeRestfulBase.class.php
index db79268bd4..5cab06099b 100644
--- a/openscholar/modules/os/modules/os_restful/plugins/restful/node/biblio/1.0/BiblioNodeRestfulBase.class.php
+++ b/openscholar/modules/os/modules/os_restful/plugins/restful/node/biblio/1.0/BiblioNodeRestfulBase.class.php
@@ -131,6 +131,10 @@ class BiblioNodeRestfulBase extends OsNodeRestfulBase {
    * Callback for getting biblio citation distribute.
    */
   public function getBibloCitationDistribute($wrapper) {
+    if (!db_table_exists('citation_distribute')) {
+      return [];
+    }
+
     $node = $wrapper->value();
     $query = db_select('citation_distribute', 'cd')
     ->condition('nid', $node->nid)
